packages/context/test/unit/context.ts

```diff
@@ -89,6 +86,13 @@ describe('Context', () => {
     it('reports an error when binding was not found', () => {
       expect(() => ctx.getBinding('unknown-key')).to.throw(/unknown-key/);
     });
+
+    it('rejects a key containing property separator', () => {
```

Comment:
Can this test be added to the one in L#26-L#29?

Reply:
I am afraid I don't understand. This test is executing `ctx.getBinding()` method, the test on lines 26-29 is executing `ctx.bind()` method. They are nested under different `describe` blocks too (we have one `describe` block for each `Context` method).

Reply:
I mean the tests seem similar in that they are both expect an error to be thrown when a property separator is found in a key. I was thinking testing them under the same `it` block makes sense:
```ts
expect(() => ctx.bind(key)).to.throw(/Binding key .* cannot contain/);
expect(() => ctx.getBinding(key)).to.throw(
        /Binding key .* cannot contain/,
      );
```

Reply:
Thanks for clarification, I understand what you mean now!

It's actually a bad practice to put multiple assertions in a single test. When one of the assertion fails, the test fails with an error and subsequent assertions are skipped. As a result, you are loosing information about which parts of your system work and which don't. Another reason is that usually the assertion failure message don't make it clear which of the multiple assertions has failed, therefore the test output does not make it clear what exactly went wrong. One has to inspect error stack trace to locate the source code line that triggered the error in order to learn which assertion failed - this adds unnecessary friction to the process.

That's why I am intentionally creating new tests for these `expect().to.throw()` checks.

Reply:
Thanks for clearing that up! Now I won't do it in the future :-)

Reply:
@b-admike sure... 😉